Separate compilation fails with object extending a trait that refers to super

`AB_1.scala`:
```scala
trait A {
  def x = 3
}
trait B extends A {
  override def x = super.x * 2
}
```
`C_2.scala`:
```scala
object C extends B {
  def main(args: Array[String]) = assert(x == args(0).toInt, x)
}
```
```bash
% dotc AB_1.scala
% dotc C_2.scala
C_2.scala:1: error: object creation impossible, since def super$x: => Int is not defined 
object C extends B {
         ^
one error found
```

`A.class` and `B.class` are identical whether or not separate compilation is used:
```java
/*
 * Decompiled with CFR 0_114.
 */
public interface A {
    default public void $init$() {
    }

    default public int x() {
        return 3;
    }
}
/*
 * Decompiled with CFR 0_114.
 */
public interface B
extends A {
    @Override
    default public void $init$() {
    }

    public /* synthetic */ int B$$super$x();

    @Override
    default public int x() {
        return this.B$$super$x() * 2;
    }
}
```

--- a/src/dotty/tools/dotc/core/tasty/TreePickler.scala
+++ b/src/dotty/tools/dotc/core/tasty/TreePickler.scala
@@ -55,11 +55,22 @@ class TreePickler(pickler: TastyPickler) {
     pickleName(TastyName.Signed(nameIndex(name), params.map(fullNameIndex), fullNameIndex(result)))
   }
 
-  private def pickleName(sym: Symbol)(implicit ctx: Context): Unit =
-    if (sym is Flags.ExpandedName)
-      pickleName(TastyName.Expanded(
-        nameIndex(sym.name.expandedPrefix), nameIndex(sym.name.unexpandedName)))
-    else pickleName(sym.name)
+  private def pickleName(sym: Symbol)(implicit ctx: Context): Unit = {
+    def encodeSuper(name: Name): TastyName.NameRef =
+      if (sym is Flags.SuperAccessor) {
+        val SuperAccessorName(n) = name
+        nameIndex(TastyName.SuperAccessor(nameIndex(n)))
+      }
+      else nameIndex(name)
+    val nameRef =
+      if (sym is Flags.ExpandedName)
+        nameIndex(
+          TastyName.Expanded(
+            nameIndex(sym.name.expandedPrefix),
+            encodeSuper(sym.name.unexpandedName)))
+      else encodeSuper(sym.name)
+    writeNat(nameRef.index)
+  }
 
   private def pickleSymRef(sym: Symbol)(implicit ctx: Context) = symRefs.get(sym) match {
     case Some(label) =>

--- a/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala
+++ b/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala
@@ -383,7 +383,6 @@ class TreeUnpickler(reader: TastyReader, tastyName: TastyName.Table) {
         case TastyName.SuperAccessor(_) => Flags.SuperAccessor
         case _ => EmptyFlags
       }
-      val superFlag = if (rawName.isInstanceOf)
       pickling.println(i"creating symbol $name at $start with flags $givenFlags")
       val flags = normalizeFlags(tag, givenFlags | nameFlags(rawName), name, isAbstractType, rhsIsEmpty)
       def adjustIfModule(completer: LazyType) =

--- a/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala
+++ b/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala
@@ -78,7 +78,7 @@ class TreeUnpickler(reader: TastyReader, tastyName: TastyName.Table) {
     case Shadowed(original) => toTermName(original).shadowedName
     case Expanded(prefix, original) => toTermName(original).expandedName(toTermName(prefix))
     case ModuleClass(original) => toTermName(original).moduleClassName.toTermName
-    case SuperAccessor(accessed) => ???
+    case SuperAccessor(accessed) => toTermName(accessed).superName
     case DefaultGetter(meth, num) => ???
   }
 
@@ -378,9 +378,14 @@ class TreeUnpickler(reader: TastyReader, tastyName: TastyName.Table) {
       val rhsIsEmpty = noRhs(end)
       if (!rhsIsEmpty) skipTree()
       val (givenFlags, annots, privateWithin) = readModifiers(end)
-      val expandedFlag = if (rawName.isInstanceOf[TastyName.Expanded]) ExpandedName else EmptyFlags
+      def nameFlags(tname: TastyName): FlagSet = tname match {
+        case TastyName.Expanded(_, original) => ExpandedName | nameFlags(tastyName(original))
+        case TastyName.SuperAccessor(_) => Flags.SuperAccessor
+        case _ => EmptyFlags
+      }
+      val superFlag = if (rawName.isInstanceOf)
       pickling.println(i"creating symbol $name at $start with flags $givenFlags")
-      val flags = normalizeFlags(tag, givenFlags | expandedFlag, name, isAbstractType, rhsIsEmpty)
+      val flags = normalizeFlags(tag, givenFlags | nameFlags(rawName), name, isAbstractType, rhsIsEmpty)
       def adjustIfModule(completer: LazyType) =
         if (flags is Module) ctx.adjustModuleCompleter(completer, name) else completer
       val sym =

--- a/tests/run/i1144/AB_1.scala
+++ b/tests/run/i1144/AB_1.scala
@@ -1,0 +1,6 @@
+trait A {
+  def x = 3
+}
+trait B extends A {
+  override def x = super.x * 2
+}

--- a/tests/run/i1144/C_2.scala
+++ b/tests/run/i1144/C_2.scala
@@ -1,0 +1,3 @@
+object Test extends B {
+  def main(args: Array[String]) = assert(x == 6, x)
+}

